src/Admin/MultisitesModelAdminExtension.php

```diff
@@ -153,27 +174,47 @@ public function getActiveSiteSessionKey() {
 		return 'Multisites_ActiveSite_For_' . get_class($this->owner);
 	}
 
-
-	/**
-	 * Get and cache an instance of the data class currently being listed
-	 *
-	 * @return DataObject
-	 **/
-	private function getListDataClass() {
-		if(!$this->listDataClass) {
-			$this->listDataClass = singleton($this->owner->getSearchContext()->getResults(array())->dataClass());
-		}		
-		return $this->listDataClass;
-	}
-
-
-	/**
-	 * Determines whether the current model being managed is MultiSitesAware
-	 *
-	 * @return boolean
-	 **/
-	private function modelIsMultiSitesAware() {
-		$model = $this->getListDataClass();
-		return $model->hasExtension('MultisitesAware') || $model instanceof SiteTree;
-	}
+    /**
+     * Get and cache an instance of the data class currently being listed
+     *
+     * @return DataObject
+     **/
+    private function getListDataClass() {
+        /**
+         * Silverstripe-recipe:4.3.0 introduces changes that broke the
+         * previous approach to retrieving the model class.
+         *
+         * This is/will be fixed in silverstripe-recipe:4.3.1
+         *
+         * We've taken this approach so that Multisites can work with
+         * SS 4.3.0 so long as you don't use the MultisitesAware extension.
+         */
+        if ($this->listDataClass === null) {
+            // only true if silverstripe-recipe >= 4.3.1
+            if ($this->owner->hasMethod('getModelClass')) {
+                $this->listDataClass = $this->owner->getModelClass();
+            }
+            // else false to bypass the MultisitesAware extension
+            else {
+                $this->listDataClass = false;
+            }
+        }
+        return $this->listDataClass;
+    }
+
+    /**
+     * Determines whether the current model being managed is MultiSitesAware
+     *
+     * @return boolean
+     **/
+    private function modelIsMultiSitesAware() {
+        $model = $this->getListDataClass();
+        if ($model) {
+            $isAware = $model::has_extension(MultisitesAware::class);
+            $isSiteTree = is_subclass_of($model, SiteTree::class);
```

Comment:
This won't work if `$model` happens to be `SiteTree`. Give `ClassInfo` a try, because I think it includes something that you can use, otherwise https://stackoverflow.com/questions/782653/checking-if-a-class-is-a-subclass-of-another.

Reply:
Ah, nice catch.